pkg/pot/lexer.go

```diff
@@ -0,0 +1,208 @@
+package pot
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type tokenType int
+
+const (
+	HeaderLanguage tokenType = iota
+	HeaderTranslator
+	HeaderPluralForms
+	MsgCtxt
+	MsgId
+	MsgStr
+	PluralMsgId
+	PluralMsgStr
+	TranslatorComment
+	ExtractedComment
+	Reference
+	Flag
+	MsgctxtPreviousContext
+	MsgidPluralPrevUntStrPlural
+	MsgidPrevUntStr
+)
+
+type Token struct {
+	Value string
+	Type  tokenType
+	Index int // plural index for msgstr with plural forms
+}
+
+var validPrefixes = []string{
+	"msgctxt ",
+	"msgid ",
+	"msgstr ",
+	"msgid_plural ",
+	"msgstr[0] ",
+	"msgstr[1] ",
+	"msgstr[2] ",
+	"msgstr[3] ",
+	"msgstr[4] ",
+	"msgstr[5] ",
+	"#: ",
+	"# ",
+	"#. ",
+	"#, ",
+	"#| msgctxt ",
+	"#| msgid_plural ",
+	"#| msgid ",
+}
+
+func Lex(r io.Reader) ([]Token, error) {
+	var tokens []Token
+
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if err := parseLine(line, &tokens); err != nil {
+			return nil, err
+		}
+	}
+
+	return tokens, nil
+}
+
+func parseLine(line string, tokens *[]Token) error {
```

Comment:
It's better for `parseLine` to return Token - removes side effects and keeps it simple.

```suggestion
func parseLine(line string) (Token, error) {
```